core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -122,13 +154,25 @@ class TimeIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writable:
       if (_entries != 0 && timestamp < lastEntry.timestamp)
         throw new IllegalStateException("Attempt to append a timestamp (%d) to slot %d no larger than the last timestamp appended (%d) to %s."
             .format(timestamp, _entries, lastEntry.timestamp, file.getAbsolutePath))
+
       // We only append to the time index when the timestamp is greater than the last inserted timestamp.
       // If all the messages are in message format v0, the timestamp will always be NoTimestamp. In that case, the time
       // index will be empty.
       if (timestamp > lastEntry.timestamp) {
+        val relativeOffset =
+          try {
```

Comment:
I think it's a little nicer to check for overflow directly rather than catching an exception.